### PR TITLE
chore: RPC client cleanup

### LINKED ISSUE #6936: op-service: RPC client code organization cleanup
Move op-node client package and base sources utils into op-service
  - [ ] `op-node/client` contains generic RPC client utils/wrappers that should be available to all services. Move all package contents.
  - [ ] `op-node/sources` contains `EthClient` and RPC batching/types utils that should be part of the op-service client package instead.
  - [ ] `op-node/testutils` contains mock bindings for the RPC interfaces, move this over too.

Split out of #6740

### PR DESCRIPTION
**Pull request by Sabnock01**
**Description**

This PR moves the following packages from `op-node` into `op-service`.

* `op-node/client`
* `op-node/sources`
* `op-node/testutils`

**Additional context**

To avoid a circular dependency, `rollupclient.go` has been moved from `sources` to `client` since `dial.go` relies on both `op-node/client` and `op-node/sources`.

Pinging @sebastianst for visibility

**Metadata**

- Fixes #6936

### CODE REVIEW

File: op-service/client/dial.go [hunk 1]
```diff
@@ -35,7 +33,7 @@ func DialEthClientWithTimeout(timeout time.Duration, log log.Logger, url string)
 
 // DialRollupClientWithTimeout attempts to dial the RPC provider using the provided URL.
 // If the dial doesn't complete within timeout seconds, this method will return an error.
-func DialRollupClientWithTimeout(timeout time.Duration, log log.Logger, url string) (*sources.RollupClient, error) {
+func DialRollupClientWithTimeout(timeout time.Duration, log log.Logger, url string) (*RollupClient, error) {
```
Comment (protolambda) [inline op-service/client/dial.go]: I think we should move the dial function out of the `client` package, and keep `RollupClient` in the `sources` package.

File: op-service/client/rollupclient.go [hunk 2]
```diff
@@ -7,16 +7,15 @@ import (
 	"github.com/ethereum/go-ethereum/common/hexutil"
 	"github.com/ethereum/go-ethereum/log"
 
-	"github.com/ethereum-optimism/optimism/op-node/client"
 	"github.com/ethereum-optimism/optimism/op-node/rollup"
 	"github.com/ethereum-optimism/optimism/op-service/eth"
 )
 
 type RollupClient struct {
```
Comment (protolambda) [inline op-service/client/rollupclient.go]: Let's keep this in the `sources` package

**Review by protolambda — COMMENTED** [review]
Nice, one thing about the `RollupClient`, but reorganization looks good to me otherwise

File: op-service/client/dial.go (hunk 1 above)
Comment (Sabnock01) [reply op-service/client/dial.go]: Moved to `op-service/dial`. Or do you have an alternate suggestion?

**Comment by protolambda** [comment]
rebased to clean up the diff with `develop`

**Review by protolambda — APPROVED** [review]

**Comment by protolambda** [comment]
rebased and squashed, no more merge conflicts now

**Review by clabby — APPROVED** [review]

**Comment by protolambda** [comment]
blocked by CI fix in #7566

**Review by roninjin10 — APPROVED** [review]

**Pull request merged**